utils/spdx/src/main/kotlin/Utils.kt

```diff
@@ -116,15 +116,42 @@ fun getLicenseText(
     licenseTextDirectories: List<File> = emptyList()
 ): String? = getLicenseTextReader(id, handleExceptions, addScanCodeLicenseTextsDir(licenseTextDirectories))?.invoke()
 
+/**
+ * Get a lambda that allows to read the full text of the license with the provided SPDX [id]. Several locations are
+ * searched for the license text: [licenseTextDirectories], the ScanCode license texts directory, the ORT's list of SPDX
+ * licenses and the resources of this module. If [handleExceptions] is enabled, the [id] may also refer to a license
+ * with an SPDX exception.
+ */
 fun getLicenseTextReader(
     id: String,
     handleExceptions: Boolean = false,
     licenseTextDirectories: List<File> = emptyList()
+): (() -> String)? {
+    return getLicenseTextReader(id, handleExceptions, licenseTextDirectories) { dir, filename ->
+        dir.resolve(filename).takeIf { it.isFile }
+    }
+}
```

Comment:
Have you considered refactoring the function to stop returning a lambda, in favour of the actual text / String?

Reply:
No, what would be the motivation to do it ? Just for code cleanup ?

BTW, do you know why it was decided to return a lambda in the first place ?

Reply:
> No, what would be the motivation to do it ?

Motivation IMO would be to have a cleaner / simpler (public) API.
The approach to do this lazily could still be kept, but by an ORT internal function.

Reply:
@fviernau I think this is already the case. The public function hides the lambda return type https://github.com/boschglobal/oss-review-toolkit/blob/ed2f67ed59a0d7970ab7d4efb0767622cef05659/utils/spdx/src/main/kotlin/Utils.kt#L113.

Reply:
This PR exposes return type ` (() -> String)? ` of `getLicenseTextReader()`, doesn't it ?

Reply:
Yes, but this is because the `LicenseTextProvider.getLicenseTextReader` also exposes this return type:
https://github.com/oss-review-toolkit/ort/blob/6a7d63dd6b9019a9bb978620b218655d0725aa42/reporter/src/main/kotlin/LicenseTextProvider.kt#L36

`getLicenseText` -> `String?`
`getLicenseTextReader` -> `(() -> String)?`

Therefore this is consistent with the current state of the implemntation.